### PR TITLE
[DependencyInjection][DomCrawler][ExpressionLanguage][Filesystem][Form] Add missing return types

### PR DESCRIPTION
**Pull request by wouterj**
| Q             | A
| ------------- | ---
| Branch?       | 6.3
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | Ref https://github.com/symfony/symfony/issues/47551
| License       | MIT
| Doc PR        | -

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
(failures are related)

**Comment by wouterj** [comment]
Related failures should be fixed now.

(not sure why Fabbot decided to start a war on me about 3 of the tens of added return PHPdocs)

**Review by stof — APPROVED** [review]

File: src/Symfony/Component/DependencyInjection/ParameterBag/FrozenParameterBag.php [hunk 1]
```diff
@@ -34,26 +34,41 @@ public function __construct(
...
+    /**
+     * @return never
+     */
     public function deprecate(string $name, string $package, string $version, string $message = 'The parameter "%s" is deprecated.')
     {
         throw new LogicException('Impossible to call deprecate() on a frozen ParameterBag.');
     }
 
+    /**
+     * @return never
```
Comment (nicolas-grekas) [inline src/Symfony/Component/DependencyInjection/ParameterBag/FrozenParameterBag.php:70]: here, I agree this is part of the contract of the FrozenParameterBag

File: src/Symfony/Component/Form/ButtonBuilder.php [hunk 2]
```diff
@@ -324,6 +324,8 @@ public function setDataLocked(bool $locked): static
     /**
      * Unsupported method.
      *
+     * @return never
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Form/ButtonBuilder.php:367]: we should also consider fixing the other methods that throw in this file

File: src/Symfony/Component/Form/Extension/Core/Type/MoneyType.php [hunk 3]
```diff
@@ -104,7 +104,7 @@ public function getBlockPrefix(): string
      * The pattern contains the placeholder "{{ widget }}" where the HTML tag should
      * be inserted
      */
-    protected static function getPattern(?string $currency)
+    protected static function getPattern(?string $currency): string
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Form/Extension/Core/Type/MoneyType.php]: why not an annotation for this one?

Comment (wouterj) [reply src/Symfony/Component/Form/Extension/Core/Type/MoneyType.php]: My bad, should be indeed.

File: src/Symfony/Component/Form/ButtonBuilder.php (hunk 2 above)
Comment (wouterj) [reply src/Symfony/Component/Form/ButtonBuilder.php:327]: To be honest, I don't get why any of the methods throwing an exception in this file exist. This class extends nothing that requires them to be in this class.

**Comment by nicolas-grekas** [comment]
Thank you @wouterj.

**Pull request merged**